crates/ruff_python_formatter/src/string/mod.rs

```diff
@@ -161,117 +125,109 @@ impl Ranged for AnyStringPart<'_> {
+}
+
+impl<'a> FormatStringPart<'a> {
+    fn new(part: &'a AnyStringPart<'a>, options: StringOptions) -> Self {
+        Self { part, options }
+    }
 }
 
-impl<'a> Format<PyFormatContext<'_>> for FormatString<'a> {
+impl Format<PyFormatContext<'_>> for FormatStringPart<'_> {
     fn fmt(&self, f: &mut PyFormatter) -> FormatResult<()> {
-        let parent_docstring_quote_style = f.context().docstring();
-        let locator = f.context().locator();
-        let result = match self.layout {
-            StringLayout::Default => {
-                if self.string.is_implicit_concatenated() {
-                    in_parentheses_only_group(&FormatStringContinuation::new(self.string)).fmt(f)
-                } else {
-                    StringPart::from_source(self.string.range(), &locator)
-                        .normalize(
-                            self.string.quoting(&locator),
-                            &locator,
-                            f.options().quote_style(),
-                            parent_docstring_quote_style,
-                        )
-                        .fmt(f)
-                }
-            }
-            StringLayout::DocString => {
-                let string_part = StringPart::from_source(self.string.range(), &locator);
-                let normalized = string_part.normalize(
-                    Quoting::CanChange,
-                    &locator,
-                    // Per PEP 8 and PEP 257, always prefer double quotes for docstrings
-                    QuoteStyle::Double,
-                    parent_docstring_quote_style,
-                );
-                docstring::format(&normalized, f)
+        match self.part {
+            AnyStringPart::String(string_literal) => {
+                string_literal.format().with_options(self.options).fmt(f)
             }
-            StringLayout::ImplicitConcatenatedStringInBinaryLike => {
-                FormatStringContinuation::new(self.string).fmt(f)
+            AnyStringPart::Bytes(bytes_literal) => bytes_literal.format().fmt(f),
+            AnyStringPart::FString(f_string) => {
+                FormatFString::new(f_string, self.options.quoting()).fmt(f)
             }
-        };
-        // TODO(dhruvmanila): With PEP 701, comments can be inside f-strings.
-        // This is to mark all of those comments as formatted but we need to
-        // figure out how to handle them. Note that this needs to be done only
-        // after the f-string is formatted, so only for all the non-formatted
-        // comments.
-        if let AnyString::FString(fstring) = self.string {
-            let comments = f.context().comments();
-            fstring.value.elements().for_each(|value| {
-                comments.mark_verbatim_node_comments_formatted(value.into());
-            });
         }
-        result
     }
 }
 
-struct FormatStringContinuation<'a> {
+pub(crate) struct FormatStringContinuation<'a> {
     string: &'a AnyString<'a>,
+    options: StringOptions,
```

Comment:
We should be able to change this to `quoting` but it requires changing `AnyStringPart` too:

Change the `StringLiteral` and `FString` variants of `AnyStringPart` and store the string literal kind and the quoting in the variants:

```rust
enum AnyStringPart<'a> {
    String {
        literal: &'a ast::StringLiteral,
        kind: StringLiteralKind,
    },
    Bytes(&'a ast::BytesLiteral),
    FString {
        string: &'a ast::FString,
        quoting: Quoting,
    },
}
```

And then implement `Format` directly on `AnyStringPart`.